platform/source/ATCmdParser.cpp

```diff
@@ -259,17 +197,24 @@ bool ATCmdParser::vrecv(const char *response, std::va_list args)
         int j = 0;
 
         while (true) {
+            // Ran out of space
+            if (j + 1 >= _buffer_size - offset) {
+                return -1;
+            }
+
             // If just peeking for OOBs, and at start of line, check
             // readability
             if (!response && j == 0 && !_fh->readable()) {
-                return false;
+                return -1;
```

Comment:
Should this actually return `0`. That's at least the behavior of C++'s scanf function.

```
On success, the function returns the number of items of the argument list successfully filled. This count can match the expected number of items or be less (even zero) due to a matching failure, a reading error, or the reach of the end-of-file.
```` - [cstdio/scanf](http://www.cplusplus.com/reference/cstdio/scanf/)